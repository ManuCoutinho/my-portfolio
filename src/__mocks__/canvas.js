HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
