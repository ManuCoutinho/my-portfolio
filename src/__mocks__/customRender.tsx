import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
