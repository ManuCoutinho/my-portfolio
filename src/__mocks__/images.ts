export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
