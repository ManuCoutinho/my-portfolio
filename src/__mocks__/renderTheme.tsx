import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
