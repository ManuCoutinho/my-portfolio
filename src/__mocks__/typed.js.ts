/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

export default class Typed {
  constructor(elementId: string, options: any) {}
  toggle(): void {}
  stop(): void {}
  start(): void {}
  destroy(): void {}
  reset(restart?: boolean): void {}
}
