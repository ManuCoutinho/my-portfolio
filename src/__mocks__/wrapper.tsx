/* eslint-disable react/display-name */
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from '__mocks__/providers/DataContext'

export const customWrapper = () => {
	return ({ children }: { children: React.ReactNode }) => (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}
