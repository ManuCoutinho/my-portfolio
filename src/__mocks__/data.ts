import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
