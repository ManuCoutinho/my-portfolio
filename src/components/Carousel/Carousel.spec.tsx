import React from 'react'
import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { Carousel } from '.'

jest.spyOn(React, 'useEffect')
jest.spyOn(React, 'useRef')
jest.spyOn(global, 'setTimeout')
jest.spyOn(global, 'clearTimeout')
jest.useFakeTimers()
describe('<Carousel />', () => {
	it('should render a title section', () => {
		render(<Carousel />)
		expect(
			screen.getByRole('heading', { name: /minhas skills/i })
		).toBeInTheDocument()
	})
	it('should render a slider with language icons', () => {
		render(<Carousel />)
		const slider = screen.getByTestId('slider')
		expect(slider).toBeInTheDocument()
		expect(slider).toHaveClass('keen-slider')
		expect(slider.firstChild).toHaveClass('keen-slider__slide')
	})
	it('should match to snapshot', () => {
		const { container } = render(<Carousel />)
		expect(container).toMatchSnapshot()
	})
})
