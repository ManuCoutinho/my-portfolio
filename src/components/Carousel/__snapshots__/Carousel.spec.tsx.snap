// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> should match to snapshot 1`] = `
<div>
  <div
    class="sc-eeb2ee64-0 igHIwV"
  >
    <h4
      class="sc-94e9b77b-0 qYMbF"
    >
      Minhas Skills:
    </h4>
    <div
      class="sc-eeb2ee64-1 jAcdEP keen-slider"
      data-testid="slider"
    >
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="akar-icons:html-fill"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="mdi:jquery"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="fa-brands:react"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="fa-brands:sass"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="ion:logo-javascript"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="mdi:nodejs"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:storybook"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:sentry"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="mdi:graphql"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="ph:figma-logo"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="simple-icons:testinglibrary"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="fa-brands:github"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="devicon-plain:docker-wordmark"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="devicon-plain:angularmaterial"
        />
      </div>
    </div>
    <div
      class="sc-eeb2ee64-1 jAcdEP keen-slider"
    >
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:typescript-alt"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:styledcomponents"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:nextjs"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:vite"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="simple-icons:jamstack"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:tailwind"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="simple-icons:mui"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="simple-icons:chakraui"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="file-icons:jest"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="devicon-plain:githubactions"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="fa-brands:bootstrap"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="akar-icons:css-fill"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="fa-brands:git-alt"
        />
      </div>
      <div
        class="sc-eeb2ee64-2 kTLTbn keen-slider__slide"
      >
        <iconify-icon
          class="sc-eeb2ee64-3 lksriL"
          classname="sc-eeb2ee64-3 lksriL"
          icon="mdi:angularjs"
        />
      </div>
    </div>
  </div>
</div>
`;
